### PR TITLE
bridge: modernise ssh-agent spawning code

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
Port the code for spawning ssh-agent to GSubprocess, merging it with the
(now-very-similar) code for spawning dbus-daemon.

We spawn ssh-agent with -D so that we have better control over its
lifetime.  In particular, the `prctl()` that we were previously doing
had no effect, because without `-D` ssh-agent `fork()`s immediately.
Closing all file descriptors also had no effect because this was already
the default behaviour of `g_spawn_sync()` and is also the default
behaviour of `GSubprocess`.

### CODE REVIEW

File: src/bridge/bridge.c [hunk 1]
```diff
@@ -174,145 +174,71 @@ send_init_command (CockpitTransport *transport,
...
+  g_assert_no_error (error);
 
-  if (!g_spawn_check_exit_status (status, &error))
+  g_autoptr(GMatchInfo) info = NULL;
+  if (!g_regex_match (regex, first_line, 0, &info))
     {
-      g_warning ("couldn't start %s: %s: %s", agent_argv[0],
-                 error->message, agent_error);
-      goto out;
+      g_warning ("output from %s didn't match expected pattern %s", argv[0], socket_pattern);
```
Comment (martinpitt) [inline src/bridge/bridge.c:220]: This is certainly a Does Not Happen™ condition, but still -- would be nice to terminate and wait the process here.

File: src/bridge/bridge.c [hunk 2]
```diff
@@ -174,145 +174,71 @@ send_init_command (CockpitTransport *transport,
...
 
-static gboolean
-have_env (const gchar *name)
+static GSubprocess *
+start_ssh_agent (void)
 {
-  const gchar *env = g_getenv (name);
-  return env && env[0];
+  const char * const cmd[] = { "ssh-agent", "-s", "-D", NULL };
+  return start_helper_process (cmd, "SSH_AUTH_SOCK=([^;]*);", "SSH_AUTH_SOCK");
```
Comment (martinpitt) [inline src/bridge/bridge.c:241]: I'm quite sure you need to escape the () here as well, just like you did in start_dbus_daemon().

**Review by martinpitt — CHANGES_REQUESTED** [review]
Nice one!

File: src/bridge/bridge.c (hunk 2 above)
Comment (allisonkarlitskaya) [reply src/bridge/bridge.c:241]: i don't use `\` anywhere, actually.

File: src/bridge/bridge.c (hunk 1 above)
Comment (allisonkarlitskaya) [reply src/bridge/bridge.c:221]: OK.  I did a force exit, and same for the similar case just above.

**Review by martinpitt — APPROVED** [review]
g_subprocess_force_exit(), oh my!

**Pull request merged**